### ISSUE TITLE
Better error message when using old opam with new root

### ISSUE THREAD

**Issue by dra27** [ocaml/opam#4266]
System opam is 2.0.4:
```
$ OPAMROOT=~/.opam21 opam switch list
Fatal error:
At /home/dra27/.opam21/config:43:0:
Invalid field depext
At /home/dra27/.opam21/config:44:0:
Invalid field depext-run-installs
At /home/dra27/.opam21/config:45:0:
Invalid field depext-cannot-install
```
`~/.opam21/config` correctly contains `opam-version: "2.0"` since the file is in 2.0 format. We need to add an extra field `opam-root-version: "2.1"` so that opam 2.1 (and opam 2.0.8) can display a more meaningful error when accidentally using the wrong binary. As we release more frequently, and encourage users to test release versions more frequently, we'll need to increase the friendliness of small mistakes 🙂

**Comment by dra27** [comment]
(this is inspired by the error message we're getting on Travis 2.0 builds, not from actually making this mistake 😉)

**Comment by AltGr** [comment]
Yup... the original intention was that the `opam-version` for this specific file should instead indicate the version of the opam root; not sure it's the best, but in this case the `opam-version` should have been upgraded...

**Comment by dra27** [comment]
Changing `opam-version` specifically here is the best fix - it means that all 2.0 clients will also consistently report the version.
